pkg/odo/cli/dev/dev.go

```diff
@@ -194,17 +198,18 @@ func (o *DevOptions) Run() error {
 	if err != nil {
 		return err
 	}
-	ceMapping, err := libdevfile.GetContainerEndpointMapping(containers)
-	if err != nil {
-		return err
-	}
+	ceMapping := libdevfile.GetContainerEndpointMapping(containers)
 	portPairs := portPairsFromContainerEndpoints(ceMapping)
 	portPairsSlice := []string{}
 	for _, v1 := range portPairs {
 		portPairsSlice = append(portPairsSlice, v1...)
 	}
+	pod, err := o.clientset.KubernetesClient.GetOnePodFromSelector(componentlabels.GetSelector(o.Context.EnvSpecificInfo.GetDevfileObj().GetMetadataName(), "app"))
```

Comment:
If you don't want to use the kubeclient in the CLI layer, you could create a function in the odo layer that would return the pod for a given componentname / appname. This way, you could also hide the selector stuff which is not the CLI concern; for example:
```
o.clientset.DevClient.GetDevPod(o.Context.EnvSpecificInfo.GetDevfileObj().GetMetadataName(), "app")
```

Reply:
Putting a function/method to find pod on cluster under `pkg/dev/dev.go` seems a bit counterproductive to me. I think it's not the place where one would expect a method to get a pod from the cluster. I have used `o.clientset.KubernetesClient.GetPodUsingComponentName` instead. That way we don't have to bother about selector either.

Reply:
The `kclient` client aims to be a pure Kubernetes library (which anyone using Kubernetes would be happy to use). Here, the component is a concept of odo, it is not a Kubernetes concept.

Reply:
The method was already there in the `kclient` interface. I have not added it. I'm not sure if you are asking me to change anything here.

Reply:
OK, so let's keep it as is for the moment. 
There are bunch of functions inside kclient that should not be there IMO, that let's reservce this for a refactoring